[Bug Report] multi-gpu tests fail

**Describe the bug**
Tests fail on multi-gpu machine.

`HookedTransformerConfig` tries to assert that `device == 'cuda'` if `n_devices > 1`. Fails if you try to load on CPU then move to `device`s

**Code example**
```
make test
```
```
FAILED tests/acceptance/test_multi_gpu.py::test_device_separation_and_cache[2] - AssertionError: n_devices > 1 is only supported on CUDA devices
FAILED tests/acceptance/test_multi_gpu.py::test_device_separation_and_cache[3] - AssertionError: n_devices > 1 is only supported on CUDA devices
FAILED tests/acceptance/test_multi_gpu.py::test_device_separation_and_cache[4] - AssertionError: n_devices > 1 is only supported on CUDA devices
ERROR tests/acceptance/test_multi_gpu.py::test_get_device_for_block_index - AssertionError: n_devices > 1 is only supported on CUDA devices
```

**System Info**
- Amazon p3.8xlarge instance with 4 GPUs.
- TransformerLens installed from source

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

--- a/transformer_lens/HookedTransformerConfig.py
+++ b/transformer_lens/HookedTransformerConfig.py
@@ -234,9 +234,6 @@ class HookedTransformerConfig:
 
         if self.n_devices > 1:
             assert (
-                self.device == "cuda"
-            ), "n_devices > 1 is only supported on CUDA devices"
-            assert (
                 torch.cuda.device_count() >= self.n_devices
             ), f"Not enough CUDA devices to support n_devices {self.n_devices}"
 

--- a/transformer_lens/utilities/devices.py
+++ b/transformer_lens/utilities/devices.py
@@ -47,8 +47,7 @@ def move_to_and_update_config(
     print_details=True,
 ):
     """
-    Wrapper around to that also changes model.cfg.device if it's a torch.device or string.
-    If torch.dtype, just passes through
+    Wrapper around `to` that also updates `model.cfg`.
     """
     if isinstance(device_or_dtype, torch.device):
         model.cfg.device = device_or_dtype.type

